Add Redis table Engine/Function (#50150)
ClickHouse/ClickHouse: docs/en/engines/table-engines/integrations/redis.md

```sql
TRUNCATE TABLE redis_table SYNC;
```

## Limitations {#limitations}

> Redis engine also support scanning query, such as `where k > xx`, but it has some limitations:

Reviewer: ```suggestion
Redis engine also supports scanning queries, such as `where k > xx`, but it has some limitations:
```

> 1. Scanning query may produce some duplicated keys in a very rare case when it is rehashing, details see [Redis Scan](https://github.com/redis/redis/blob/e4d183afd33e0b2e6e8d1c79a832f678a04a7886/src/dict.c#L1186-L1269)

Reviewer: ```suggestion
1. Scanning query may produce some duplicated keys in a very rare case when it is rehashing. See details in [Redis Scan](https://github.com/redis/redis/blob/e4d183afd33e0b2e6e8d1c79a832f678a04a7886/src/dict.c#L1186-L1269)
```

> 2. During the scanning keys could be created and deleted, so the resulting dataset can not represent a valid point in time.

Reviewer: ```suggestion
2. During the scanning, keys could be created and deleted, so the resulting dataset can not represent a valid point in time.
```
